### PR TITLE
H2STORE test cases

### PR DESCRIPTION
**Pull request by svenn-t**
Depends on https://github.com/OPM/opm-common/pull/3445 and https://github.com/OPM/opm-simulators/pull/4528 
These test cases are the same as for CO2STORE, but with H2STORE keyword enabled.

### CODE REVIEW

File: H2STORE_GW.DATA
Comment (goncalvesmachadoc) [inline H2STORE_GW.DATA:14]: CO2STORE -> H2STORE

File: H2STORE_GW.DATA
Comment (goncalvesmachadoc) [inline H2STORE_GW.DATA:6]: 2021 -> 2023

File: H2STORE_GW.DATA
Comment (goncalvesmachadoc) [inline H2STORE_GW.DATA:93]: 0.7/ 35-40g/l  -> 35-40g /kg -> 0.63-0.72 mol/g
->
0.7/ -- 35-40g/l  -> 35-40g /kg -> 0.63-0.72 mol/g

File: H2STORE_GW.DATA
Comment (bska) [inline H2STORE_GW.DATA:1]: This file (`H2STORE_GW.DATA`) appears to have been added **outside** of the new `h2store` directory&mdash;i.e., in the top-level `opm-tests` directory.  Is that intentional?  If not I'd prefer if it were moved into the `h2store` directory.

File: H2STORE_GW.DATA
Comment (svenn-t) [reply H2STORE_GW.DATA]: Corrected, thanks!

File: H2STORE_GW.DATA
Comment (svenn-t) [reply H2STORE_GW.DATA]: Corrected, thanks!

File: H2STORE_GW.DATA
Comment (svenn-t) [reply H2STORE_GW.DATA]: Corrected, thanks!

File: H2STORE_GW.DATA
Comment (svenn-t) [reply H2STORE_GW.DATA:1]: Sorry, that was not intentional. Moved it inside h2store directory now.

**Comment by totto82** [comment]
@bska and @goncalvesmachadoc Can this be merged now? I would like to get this in before merging the update on the H2STORE option to make sure we have proper tests for the new functionality.

**Review by bska — COMMENTED** [review]
> Can this be merged now?

I don't see why not as most of this is straightforward.  I'll let @goncalvesmachadoc have the final say, though, because I don't know enough about the physics here to know if I'm missing some subtle detail.

File: h2store/H2STORE.DATA
Comment (goncalvesmachadoc) [inline h2store/H2STORE.DATA:7]: Please add short description of the case.

File: h2store/H2STORE.DATA
Comment (goncalvesmachadoc) [inline h2store/H2STORE.DATA:42]: Remove NOECHO.

File: h2store/H2STORE_DIFFUSIVE.DATA
Comment (goncalvesmachadoc) [inline h2store/H2STORE_DIFFUSIVE.DATA:7]: Please add short description of the case.

File: h2store/H2STORE_DIFFUSIVE.DATA
Comment (goncalvesmachadoc) [inline h2store/H2STORE_DIFFUSIVE.DATA:43]: Remove NOECHO.

File: h2store/H2STORE_DISGAS.DATA
Comment (goncalvesmachadoc) [inline h2store/H2STORE_DISGAS.DATA:7]: Please add short description of the case.

File: h2store/H2STORE_DISGAS.DATA
Comment (goncalvesmachadoc) [inline h2store/H2STORE_DISGAS.DATA:43]: Remove NOECHO.

File: h2store/H2STORE_ENERGY.DATA
Comment (goncalvesmachadoc) [inline h2store/H2STORE_ENERGY.DATA:7]: Please add short description of the case.

File: h2store/H2STORE_ENERGY.DATA
Comment (goncalvesmachadoc) [inline h2store/H2STORE_ENERGY.DATA:45]: Remove NOECHO.

File: h2store/H2STORE_GW.DATA
Comment (goncalvesmachadoc) [inline h2store/H2STORE_GW.DATA:49]: Remove NOECHO.

File: h2store/H2STORE_GW.DATA
Comment (goncalvesmachadoc) [inline h2store/H2STORE_GW.DATA:180]: Is it necessary to have a longer run for this case in particular? All other cases are 30*1.

**Review by goncalvesmachadoc — CHANGES_REQUESTED** [review]

**Pull request merged**
